pkg/credentialprovider/config.go

```diff
@@ -122,14 +123,27 @@ func ReadDockerConfigJSONFile(searchPaths []string) (cfg DockerConfig, err error
 	if len(searchPaths) == 0 {
 		searchPaths = DefaultDockerConfigJSONPaths()
 	}
+	return ReadSpecificDockerConfigJsonFile(searchPaths, configJsonFileName)
+}
 
+// ReadDockerConfigJsonTransformationFile attempts to read a legacy .dockerconfigjson file from the given paths.
+// if searchPaths is empty, the default paths are used.
+func ReadDockerConfigJsonFileUsedInSecret(searchPaths []string) (cfg DockerConfig, err error) {
+	if len(searchPaths) == 0 {
+		searchPaths = DefaultDockercfgPaths()
+	}
+	return ReadSpecificDockerConfigJsonFile(searchPaths, configJsonFileNameInSecret)
+}
```

Comment:
@bparees  `ReadSpecificDockerConfigJsonFile `is called by `ReadDockerConfigJsonFileUsedInSecret `and `ReadDockerConfigJSONFile`, the main distinction between two function is default searching path and filename. That's my intention to wrap this function.  In function `ReadDockerConfigFile `I think we can't call this function directly, because if we set nil to searchpath, it will do noting for kube or if we set some specific search paths we have to refactor `ReadDockerConfigFile`.